hilti/toolchain/include/ast/types/library.h

```diff
@@ -26,17 +28,19 @@ class Library : public UnqualifiedType {
     bool isMutable() const final { return true; }
 
     node::Properties properties() const final {
-        auto p = node::Properties{{"cxx_name", _cxx_name}};
+        auto constness = (_constness == Constness::Const ? "true" : "false");
+        auto p = node::Properties{{"const", constness}, {"cxx_name", _cxx_name}};
         return UnqualifiedType::properties() + std::move(p);
     }
 
-    static auto create(ASTContext* ctx, const std::string& cxx_name, Meta meta = {}) {
-        return ctx->make<Library>(ctx, cxx_name, std::move(meta));
+    static auto create(ASTContext* ctx, Constness const_, const std::string& cxx_name, Meta meta = {}) {
+        return ctx->make<Library>(ctx, const_, cxx_name, std::move(meta));
     }
```

Comment:
The constructor takes `cxx_name` by value, let's do the same here and `move` the value,

```suggestion
    static auto create(ASTContext* ctx, Constness const_, std::string cxx_name, Meta meta = {}) {
        return ctx->make<Library>(ctx, const_, std::move(cxx_name), std::move(meta));
    }
```

With that we should also adjust the signature in `node-factory.h`.